cmake/FindHEIF.cmake

```diff
@@ -0,0 +1,50 @@
+# - Try to find the libheif library
+# Once done this will define
+#
+#  HEIF_FOUND - system has heif
+#  HEIF_INCLUDE_DIRS - the heif include directories
+#  HEIF_LIBRARIES - the libraries needed to use heif
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+if (NOT WIN32)
+    include(LibFindMacros)
+    libfind_pkg_check_modules(HEIF_PKGCONF libheif)
+
+    find_path(HEIF_INCLUDE_DIR
+        NAMES libheif/heif.h
+        HINTS ${HEIF_PKGCONF_INCLUDE_DIRS} ${HEIF_PKGCONF_INCLUDEDIR}
+        PATH_SUFFIXES heif
+    )
+
+    find_library(HEIF_LIBRARY
+        NAMES heif
+        HINTS ${HEIF_PKGCONF_LIBRARY_DIRS} ${HEIF_PKGCONF_LIBDIR}
+    )
+
+    set(HEIF_PROCESS_LIBS HEIF_LIBRARY)
+    set(HEIF_PROCESS_INCLUDES HEIF_INCLUDE_DIR)
+    libfind_process(HEIF)
+
+else()
+    find_path(HEIF_INCLUDE_DIR
+        NAMES heif.h
+    )
+
+    find_library (
+        HEIF_LIBRARY
+        NAMES libheif libheif
+        DOC "Libraries to link against for HEIF Support"
+    )
+
```

Comment:
Here we should add a check to detect if the library has not been found:

```
if(NOT HEIF_LIBRARY)
  message(FATAL_ERROR "HEIF library not found")
endif()
```

In fact, after installing the packages in my ubuntu-18.04:

```
(conan) luis@desktopXubuntu:/media/linuxDev/programming/builds/exiv2Release$ dpkg -l | grep heif
ii  libheif-dev:amd64                     1.1.0-2                               amd64        ISO/IEC 23008-12:2017 HEIF file format decoder - development files
ii  libheif1:amd64                        1.1.0-2                               amd64        ISO/IEC 23008-12:2017 HEIF file format decoder - shared library
```

This code is not able to find properly the library, and I am getting liking errors.

Reply:
These Ubuntu packages are very old.
https://github.com/strukturag/libheif/releases

Reply:
Good to know. I'll give a try with those versions. I could also explore the possibility to create a conan recipe for that library after merging this branch into our project.

Now that you mentioned the versions, we should also add something in this `FindHEIF.cmake` file related to the minimum required version of the library. Let's take my previous experience ... if you know for sure that version 1.1.0 will not work, CMake should complain about it.